File: lib/saga/saga_activity_deposit.ex
defmodule Saga.Activity.Deposit do
  use TemporalSdk.Activity
  @moduledoc false

  @impl true
  def execute(_context, _input) do
    Process.sleep(1_000)
    ["deposit"]
  end

  @impl true
  def handle_heartbeat(_context), do: :heartbeat

  @impl true
  def handle_cancel(%{cancel_requested: true}), do: {:cancel, ["details"]}
  def handle_cancel(_context), do: :ignore
end

File: src/saga/saga_activity_deposit.erl
-module(saga_activity_deposit).

% elp:ignore W0012 W0040
-moduledoc false.

-export([
    execute/2,
    handle_heartbeat/1,
    handle_cancel/1
]).

-include_lib("temporal_sdk/include/activity.hrl").

execute(_Context, _Input) ->
    timer:sleep(100),
    ["deposit"].

handle_heartbeat(_Context) -> heartbeat.

handle_cancel(#{cancel_requested := true}) -> {cancel, ["details"]};
handle_cancel(_Context) -> ignore.
